### DISCUSSION TITLE
Build Orders & Serialised Items

### DISCUSSION THREAD

**Discussion by guidoBlee** [inventree/InvenTree#8642] [Q&A] [answered]
Hi, 

This is my first post on this forum. I have just deployed Inventree for a SMB in the UK who make composite materials. I love the project, and think there's tonnes of useful features. I've already deployed a plugin to print to an internal web endpoint and am excited to do more. 

I have a couple of questions about build orders: 

1. It is somewhat confusing to me that you set up a build order with a quantity of a part you want to make - this defines the stock allocation, ok so far. However - this has no bearing on the actual Build Outputs, not even to populate a default Incomplete Item. Is there a reason for this? 

Where I am struggling, however, is that a part has to be trackable (serialised) to have part tests associated with it. However, a Build Output producing a serialised part cannot have Build Outputs with non-integer quantities. This is difficult when my stock is not counted by units but by area (metres squared). 

What I have so far is:

- There is a panel Part with the correct Bill of Materials required to make one panel of, say, 4.2m^2
- I set up a build order with a quantity of one, to make a panel.
- It allocates the stock required to make one single panel. 
- I create a build output of 4.2m^2 
- ERROR - serialised parts must have an integer quantity. 

Is there a fix / workaround for this?

2. I cannot work out the 'Delete on Deplete' option. I would like to be able to see full stock histories to track and query past panels we have manufactured, so I figure I should have this option disabled - I have confirmed it is disabled globally. I struggle to see what actual behaviour the option is triggering though, as I have enabled it for a part on the demo database and I can still see it after consuming it in a build order. 

Can anyone tell me what actual logic the 'Delete on Deplete' option is controlling? I have dug through the docs and the source code but not found any actions taken on the basis of the option's state. 

Any and all help is much appreciated.
Guido

**Comment by SchrodingersGat** [comment, marked as answer]
Hey @guidoBlee thanks for the kind feedback :)

### Testable vs Trackable Parts

In the upcoming `0.17.0` release, we have separated the concepts of "testable" and "trackable":

- https://docs.inventree.org/en/latest/part/part/#testable
- https://docs.inventree.org/en/latest/part/part/#trackable

(Note that these docs are for the *latest* version, not the *stable* version).

So, you can define tests independent of the serial number requirement. The `0.17.0` version *should* be out before Christmas.

### Delete on Deplete

This controls the behaviour of the particular stock item when the quantity depletes to zero. If enabled, it will remove the stock item (and all associated history) from the database

**Reply by guidoBlee** [reply]
Thanks for the fast answer @SchrodingersGat, this all sounds good. I look forward to getting the update.
